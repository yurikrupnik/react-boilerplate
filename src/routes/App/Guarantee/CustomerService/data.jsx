const data = {
    header: '24/7 Customer Service',
    text: 'Our ravely-reviewed Customer Service team of non-commissioned diamond and jewelry experts is here for you 24/7.',
    items: [
        {
            type: 'general',
            icon: 'call',
            header: 'Call Us',
            mobileLabel: 'Call',
            path: '',
            text: 'USA: 877 826 9866, outside USA: +1 301 631 1414'
        }, {
            type: 'general',
            icon: 'chat',
            header: 'Chat Now',
            mobileLabel: 'Chat',
            path: '',
            text: 'Get instant answers to your questions'
        }, {
            type: 'general',
            icon: 'email',
            header: 'Email Us',
            mobileLabel: 'Email',
            path: 'mailto:social@jamesallen.com',
            text: 'All questions will be answered within two business days.'
        }, {
            type: 'desktop',
            icon: 'realTime',
            header: 'Real-Time Diamond Inspection',
            mobileLabel: '',
            path: '/diamond-consultation',
            text: 'We offer an interactive Real-time Diamond Inspection service, the first of its kind in'
                + ' the industry. Our experts will share their screen with you, and walk you through the attributes'
                + ' of your favorite diamonds in Super Zoom magnification. Ask questions, mark points of interest on the '
                + 'diamond itself, and review everything from the specs to the grading certificate together.'
        }, {
            type: 'desktop',
            icon: 'message',
            header: 'Send a Message',
            mobileLabel: '',
            path: '',
            text: 'All questions will be answered within two business days'
        }, {
            type: 'mobile',
            icon: 'skype',
            mobileLabel: 'Skype',
            path: '',
        }
    ]
};

export default data;
