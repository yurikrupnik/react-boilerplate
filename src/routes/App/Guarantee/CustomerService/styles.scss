.customerServiceContainer {
margin-bottom: 25px;
}

.customerServiceHeader {
  margin: 0;
  font-size: 26px;
  color: #ff8080;
  padding-bottom: 22px;
}

.headerItem {
  font-size: 16px;
  text-decoration: underline;
}

.headerItem a {
  color: black;
}

.itemContainer {
  margin-bottom: 20px;
}
