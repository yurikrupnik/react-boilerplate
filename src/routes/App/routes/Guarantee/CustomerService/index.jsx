import React from 'react';
// import PropTypes from 'prop-types';

const CustomerService = (props) => (

    <div>
        <h2>24/7 Customer Service</h2>
        <p>
            Our ravely-reviewed Customer Service team of non-commissioned
            diamond and jewelry experts is here for you 24/7.
        </p>
        <div className="row">
            <div className="col-xs-1">
                icon
            </div>
            <div className="col-xs-11">
                <div>
                    Call Us
                </div>
                <div>
                    USA: 877 826 9866, outside USA: +1 301 631 1414
                </div>
            </div>
        </div>
        <div className="row">
            <div className="col-xs-1">
                icon
            </div>
            <div className="col-xs-11">
                <div>
                    Cgat Now
                </div>
                <div>
                    Get instant answers to your questions.
                </div>
            </div>
        </div>
        <div className="row">
            <div className="col-xs-1">
                icon
            </div>
            <div className="col-xs-11">
                <div>
                    Email us
                </div>
                <div>
                    All questions will be answered within two business days.
                </div>
            </div>
        </div>
        <div className="row">
            <div className="col-xs-1">
                icon
            </div>
            <div className="col-xs-11">
                <div>
                    Real-Time Diamond Inspection
                </div>
                <div>
                    We offer an interactive Real-time Diamond Inspection service,
                    the first of its kind in the industry. Our experts will share their
                    screen with you, and walk you through the attributes of your favorite
                    iamonds in Super Zoom magnification. Ask questions, mark points of interest
                    on the diamond itself, and review everything from the specs to the
                    grading certificate together.
                </div>
            </div>
        </div>
        <div className="row">
            <div className="col-xs-1">
                icon
            </div>
            <div className="col-xs-11">
                <div>
                    Send a Message
                </div>
                <div>
                    All questions will be answered within two business days.
                </div>
            </div>
        </div>
        <div>
            <div className="row">
                <div className="col-xs-4">
                    <div>
                        name
                    </div>
                    <input type="text" />
                </div>
                <div className="col-xs-4">
                    <div>
                        Email
                    </div>
                    <input type="text" />
                </div>
                <div className="col-xs-4">
                    <div>
                        Phone
                    </div>
                    <input type="text" />
                </div>
            </div>
            <div>
                <textarea style={{width: '100%'}}></textarea>
            </div>
            <div>
                <button>
                    Send
                </button>
            </div>
        </div>
    </div>
);


// CustomerService.propTypes = {
//     match: PropTypes.shape({}).isRequired
// };

export default CustomerService;
