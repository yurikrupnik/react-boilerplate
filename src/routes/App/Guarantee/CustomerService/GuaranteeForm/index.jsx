import React, { Component } from 'react';
import styles from './styles.scss';

class GuaranteeForm extends Component {
    render() {
        return (
            <div className={styles.mainContainer}>
                <div className={styles.inputsContainer}>
                    <div>
                        <div>
                            Name
                        </div>
                        <input type="text" />
                    </div>
                    <div>
                        <div>
                            Email
                        </div>
                        <input type="text" />
                    </div>
                    <div className={styles.test}>
                        <div>
                            Phone (optional)
                        </div>
                        <input type="text" />
                    </div>
                </div>
                <div className={styles.textareaContainer}>
                    <div>
                        Message
                    </div>
                    <textarea />
                </div>
                <div className={styles.btnContainer}>
                    <button>
                        Send
                    </button>
                </div>
            </div>
        );
    }
}

export default GuaranteeForm;
