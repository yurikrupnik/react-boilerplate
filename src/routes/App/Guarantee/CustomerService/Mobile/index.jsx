/* eslint-disable linebreak-style */
import React from 'react';
import styles from './styles.scss';
import HeaderMobile from '../../HeaderMobile';
import data from '../data';

const CustomerService = () => (
    <div className={styles.rootContainer}>
        <HeaderMobile text={data.header} />
        <p>
            {data.text}
        </p>
        <div className={styles.servicesContainer}>
            <a className={styles.serviceItem}>
                <div className={styles.imgService}>
                    <h3>
                        LOGO
                    </h3>
                </div>
                Call
            </a>
            <a className={styles.serviceItem}>
                <div className={styles.imgService}>
                    <h3>
                        LOGO
                    </h3>
                </div>
                Skype
            </a>
            <a className={styles.serviceItem}>
                <div className={styles.imgService}>
                    <h3>
                        LOGO
                    </h3>
                </div>
                Chat
            </a>
            <a className={styles.serviceItem}>
                <div className={styles.imgService}>
                    <h3>
                        LOGO
                    </h3>
                </div>
                Email
            </a>
        </div>

    </div>
);

export default CustomerService;
