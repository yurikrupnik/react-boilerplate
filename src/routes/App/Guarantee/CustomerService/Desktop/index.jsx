/* eslint-disable linebreak-style */
import React from 'react';
import { Link } from 'react-router-dom';
import styles from './styles.scss';
import GuaranteeForm from './GuaranteeForm/index';
import data from '../data';

const CustomerService = () => {
    const { header, text, items } = data;
    return (
        <div className={styles.root}>
            <h1 className={styles.header}>
                {header}
            </h1>
            <p>
                {text}
            </p>
            <div>
                {items.map(item => (
                    (item.type === 'general' || item.type === 'desktop')
                     && (
                         <div className={`row ${styles.itemContainer}`} key={item.header}>
                             <span className={`col-xs-1 ${styles.icon} ${styles[item.icon]}`} />
                             <div className="col-xs-11">
                                 {item.path
                                     ? (
                                         <div className={styles.headerItem}>
                                             {item.header === 'Email Us'
                                                 ? (
                                                     <a ajaxmode="disabled" href={item.path}>
                                                         {item.header}
                                                     </a>
                                                 )
                                                 : (
                                                     <Link to={item.path}>
                                                         {item.header}
                                                     </Link>
                                                 )
                                             }
                                         </div>
                                     )
                                     : (
                                         <div>
                                             {item.header}
                                         </div>
                                     )
                                 }
                                 <div>
                                     {item.text}
                                 </div>
                             </div>
                         </div>
                     )))}
            </div>
            <div className="row">
                <GuaranteeForm className="col-xs-11" />
            </div>
        </div>
    );
};


export default CustomerService;
