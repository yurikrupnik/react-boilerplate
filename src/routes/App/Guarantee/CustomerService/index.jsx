import Mobile from './Mobile/index';
import Desktop from './Desktop/index';

export default [Mobile, Desktop];
