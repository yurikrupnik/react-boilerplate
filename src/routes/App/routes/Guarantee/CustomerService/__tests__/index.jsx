import React from 'react';
import CustomerService from '../Desktop/index';

const {
    describe,
    test,
    expect,
    shallow
} = global;

describe('CustomerService', () => {
    test('render CustomerService', () => {
        const wrapper = shallow(
            <CustomerService />
        );
        expect(wrapper).toMatchSnapshot();
        // expect(wrapper.find('div').length).toBe(1);
    });
});
