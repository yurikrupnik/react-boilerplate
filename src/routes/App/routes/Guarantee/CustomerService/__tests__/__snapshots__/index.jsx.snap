// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerService render CustomerService 1`] = `
<div
  className="customerServiceContainer"
>
  <h1
    className="customerServiceHeader"
  >
    24/7 Customer Service (DESKTOP)
  </h1>
  <p>
    Our ravely-reviewed Customer Service team of non-commissioned diamond and jewelry experts is here for you 24/7.
  </p>
  <div>
    <div
      className="row itemContainer"
      key="Call Us"
    >
      <div
        className="col-xs-1"
      >
        icon
      </div>
      <div
        className="col-xs-11"
      >
        <div>
          Call Us
        </div>
        <div>
          USA: 877 826 9866, outside USA: +1 301 631 1414
        </div>
      </div>
    </div>
    <div
      className="row itemContainer"
      key="Chat Now"
    >
      <div
        className="col-xs-1"
      >
        icon
      </div>
      <div
        className="col-xs-11"
      >
        <div>
          Chat Now
        </div>
        <div>
          Get instant answers to your questions
        </div>
      </div>
    </div>
    <div
      className="row itemContainer"
      key="Email Us"
    >
      <div
        className="col-xs-1"
      >
        icon
      </div>
      <div
        className="col-xs-11"
      >
        <div
          className="headerItem"
        >
          <a
            ajaxmode="disabled"
            href="mailto:social@jamesallen.com"
          >
            Email Us
          </a>
        </div>
        <div>
          All questions will be answered within two business days.
        </div>
      </div>
    </div>
    <div
      className="row itemContainer"
      key="Real-Time Diamond Inspection"
    >
      <div
        className="col-xs-1"
      >
        icon
      </div>
      <div
        className="col-xs-11"
      >
        <div
          className="headerItem"
        >
          <Link
            replace={false}
            to="/diamond-consultation"
          >
            Real-Time Diamond Inspection
          </Link>
        </div>
        <div>
          We offer an interactive Real-time Diamond Inspection service, the first of its kind in the industry. Our experts will share their screen with you, and walk you through the attributes of your favorite diamonds in Super Zoom magnification. Ask questions, mark points of interest on the diamond itself, and review everything from the specs to the grading certificate together.
        </div>
      </div>
    </div>
    <div
      className="row itemContainer"
      key="Send a Message"
    >
      <div
        className="col-xs-1"
      >
        icon
      </div>
      <div
        className="col-xs-11"
      >
        <div>
          Send a Message
        </div>
        <div>
          All questions will be answered within two business days
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <GuaranteeForm
      className="col-xs-11"
    />
  </div>
</div>
`;
